src/constraints/constraint_common.jl

```diff
@@ -70,11 +70,22 @@ past_units_on_indices(m, param, u, s_path, t) = _past_indices(m, units_on_indice
 
 function _past_indices(m, indices, param, s_path, t; kwargs...)
     look_behind = maximum(maximum_parameter_value(param(; kwargs..., stochastic_scenario=s, t=t)) for s in s_path)
+    
+    function convert_to_days(duration)
+        if isa(duration, Year)
+            return Day(Dates.value(duration) * 365)  # approximate conversion
+        elseif isa(duration, Month)
+            return Day(Dates.value(duration) * 30)   # approximate conversion
+        else
+            return duration                          # do nothing
+        end
+    end
+
```

Comment:
Nice work @datejada - a few comments.
- I think you might run into the same issues with connection and storage lifetimes, so probably you will also need to call this function there as well.
- The other thing is just style, in Julia you have multiple dispatch which allows you to avoid those if else blocks to switch the variable type. You may have seen it, it looks something like this:
```
convert_to_days(duration::Year) = Day(Dates.value(duration) * 365)
convert_to_days(duration::Month) = Day(Dates.value(duration) * 30)
convert_to_days(duration) = duration
```
You may want to try it instead, it's faster too although it doesn't look like this code is performance-critical.
- You have chosen 365 and 30 as the multipliers which are the median but I wonder if you need to take something else instead, like the mean or the maximum. We are using the result to call isless, so si the result of isless correct in all cases?
- Unit-tests?

Reply:
Hi @manuelma 

Thank you for your comments; they are constructive 😄 So:

The new function is inside the `_past_indices` function, which is the same one used for connections and storage. So, we will catch all the cases here.
- Awesome, I will change it like this 😉 it looks more Julia.
- Good point! So, the best approximation to cover the cases will be taking the maximum since we look behind the lifetime in the index t that is in the `isless`, meaning that overestimating (i.e., 366 and 31) will guarantee to cover the lifetime to determine the `weight` value.
- I don't think we need extra tests for now since the new economic parameters branch #909 will test this situation (and hopefully, after the latest changes, it will pass the tests and can be merged soon)